Development/sdp/sdp_grammar.cpp

```diff
@@ -68,6 +68,22 @@ namespace sdp
             return web::json::value(v);
         }
 
+        inline std::string jns2s(const web::json::value& v)
+        {
+            if (v.is_number()) return jn2s(v);
+            else return js2s(v);
+        }
+        inline web::json::value digits2jns(const std::string& s)
+        {
+            if (!std::all_of(s.begin(), s.end(), ::isdigit)) throw sdp_parse_error("expected a sequence of digits");
+
+            uint64_t v;
+            std::istringstream is(s);
+            is >> v;
```

Comment:
This doesn't seem to solve the problem that 1*DIGITS strings that begin with leading zeroes are not roundtrippable. If you're going to store these as strings do you want to try to solve that too?

Cf. https://github.com/sony/nmos-cpp/commit/062be351ebd58565f91ba1a51c858b9cf19a2d0a#diff-159bd613b3d4d427540279e8cb51487660592ac0bf8634ba88a92dabb7527470R61

Reply:
For allowing leading zeroes, I think I do the following will do?
```
inline web::json::value digits2jns(const std::string& s)
{
    if (!std::all_of(s.begin(), s.end(), ::isdigit)) throw sdp_parse_error("expected a sequence of digits");
    return s2js(s);
}
```

Reply:
Yeah, I think so!